src/PhpBrew/VariantBuilder.php

```diff
@@ -315,34 +315,27 @@ public function __construct()
             return;
         };
 
-        $this->variants['curl'] = function (Build $build, $prefix = null) {
-            if ($prefix) {
-                return "--with-curl=$prefix";
-            }
-
-            if ($prefix = Utils::findIncludePrefix('curl/curl.h')) {
-                return "--with-curl=$prefix";
-            }
+        $this->variants['curl'] = function (Build $build, $val = null) {
+            $prefix = Utils::findPrefix(array(
+                new BrewPrefixFinder('curl'),
+                new PkgConfigPrefixFinder('curl'),
+                new IncludePrefixFinder('curl/curl.h'),
+            ));
 
-            if ($prefix = Utils::getPkgConfigPrefix('libcurl')) {
-                return "--with-curl=$prefix";
-            }
+            if ($build->compareVersion('7.4') < 0) {
```

Comment:
The `+curl` variant seems to work fine on Linux regardless of the PHP version. Is it specific to macOS and what exactly has changed in PHP in that regard?

Reply:
Please see the big comment below.